File: example/ios/native-navigation/ExampleBridge.m
//  Created by react-native-create-bridge

// import RCTBridgeModule
#import <React/RCTBridgeModule.h>

// Export a native module
// https://facebook.github.io/react-native/docs/native-modules-ios.html#exporting-swift
@interface RCT_EXTERN_MODULE(ExampleBridge, NSObject)

- (dispatch_queue_t)methodQueue {
    return dispatch_get_main_queue();
}

// Export methods to a native module
// https://facebook.github.io/react-native/docs/native-modules-ios.html#exporting-swift
RCT_EXTERN_METHOD(exampleMethod:(NSString *)screenName)

@end

File: example/ios/native-navigation/ExampleBridge.swift
//  Created by react-native-create-bridge

import Foundation
import NativeNavigation

@objc(ExampleBridge)
class ExampleBridge : NSObject {
  fileprivate let coordinator: ReactNavigationCoordinator
    
    override init() {
        coordinator = ReactNavigationCoordinator.sharedInstance
    }
    
  // Export constants to use in your native module
  @objc func constantsToExport() -> [String : Any]! {
    return ["EXAMPLE_CONSTANT": "example"]
  }

  // Implement methods that you want to export to the native module
  @objc func exampleMethod(_ screenName: String) {
    // write method here
    guard let topVc = coordinator.topViewController() else {
        return
    }
    
    let vc = TestViewController(screenName: screenName)
    topVc.navigationController?.pushViewController(vc, animated: true)
  }
}
